apps/web/src/services/cost-explorer/budget/shared/BudgetDataSourceSelect.vue

```diff
@@ -0,0 +1,113 @@
+    isAllValid,
+} = useFormValidator({
+    selectedDataSource: null as string|null,
+}, {
+    selectedDataSource(value: string|null) { return !!value; },
+});
+
+const state = reactive({
+    dataSourceItems: [] as MenuItem[],
+});
+
+watch([() => selectedDataSource.value, () => isAllValid.value], ([dataSource, isValid]) => {
+    emit('update', dataSource, isValid);
+}, { immediate: true });
+
+const fetchDataSource = async () => {
+    try {
+        const { results } = await SpaceConnector.clientV2.costAnalysis.dataSource.list();
+
+        const dataSourceItems: MenuItem[] = results.map((dataSource:DataSourceInfoModel) => ({
+            name: dataSource.data_source_id,
+            label: dataSource.name,
+            imageUrl: 'https://spaceone-custom-assets.s3.ap-northeast-2.amazonaws.com/console-assets/icons/aws-ec2.svg',
+        }));
+
+        state.dataSourceItems = dataSourceItems;
+    } catch (e) {
+        ErrorHandler.handleError(e);
+        state.dataSourceItems = [];
+        setForm('selectedDataSource', null);
+    }
+};
+
+(async () => {
+    await fetchDataSource();
+    setForm('selectedDataSource', state.dataSourceItems[0]?.name);
+})();
+
+</script>
+
+<template>
+    <p-field-group :label="$t('BILLING.COST_MANAGEMENT.BUDGET.FORM.BASE_INFO.DATA_SOURCE')"
+                   :invalid="!selectedDataSource"
+                   required
+                   class="budget-data-source-select-field"
+    >
+        <p-select-dropdown class="data-source-dropdown"
+                           :items="state.dataSourceItems"
+                           :selected="selectedDataSource"
+                           @update:selected="setForm('selectedDataSource', $event)"
+        >
+            <template #default="{ item }">
+                <!--TODO: Currency should be changed to real data.-->
+                <div class="selected-input">
+                    <p-lazy-img :src="item.imageUrl"
+                                width="1rem"
+                                height="1rem"
+                    /><span>{{ item.label }}</span> <span class="selected-item-postfix">(Currency: ₩KRW)</span>
+                </div>
+            </template>
+            <template #menu-item--format="{item}">
+                <div class="menu-item">
+                    <!--TODO: Currency should be changed to real data.-->
+                    <span>{{ item.label }}</span> <span class="selected-item-postfix">(Currency: ₩KRW)</span>
+                </div>
+            </template>
+        </p-select-dropdown>
+    </p-field-group>
+</template>
+
+
+<style lang="postcss" scoped>
+.budget-data-source-select-field {
+    width: 30rem;
+    .data-source-dropdown {
+        width: 100%;
+        .selected-input {
+            @apply flex items-center gap-1;
+        }
+        .selected-item-postfix {
+            @apply text-gray-400;
+        }
+    }
+}
+
+@screen mobile {
+    .budget-data-source-select-field {
+        width: 100%;
+    }
+}
```

Comment:
issue:
<img width="584" alt="image" src="https://github.com/cloudforet-io/console/assets/26986739/42d59b86-6514-4b64-a5a0-ddd2b92d020b">

In long contents case, it overflows.. T.T
Please check it.